plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixTypeMapping.java

```diff
@@ -344,6 +344,22 @@ public void testChar()
                 .execute(getQueryRunner(), phoenixCreateAndInsert("tpch.test_char"));
     }
 
+    @Test
+    public void testBinary()
+    {
+        // Not testing max length (2147483647) because it leads to 'Requested array size exceeds VM limit'
+        SqlDataTypeTest.create()
+                .addRoundTrip("binary(1)", "NULL", VARBINARY, "X'00'") // NULL stored as zeros
+                .addRoundTrip("binary(10)", "DECODE('', 'HEX')", VARBINARY, "X'00000000000000000000'") // empty stored as zeros
```

Comment:
Should these be considered bugs?

i.e. Would a query like `SELECT * FROM tbl WHERE col IS NULL` return rows where the value is `X'00'`?

Reply:
No, it's not bugs. Added another test case to ensure the behavior.